src/zjs_callbacks.c

```diff
@@ -315,6 +334,20 @@ void zjs_remove_callback(int32_t id)
     }
 }
 
+void zjs_signal_callback_args(int32_t id, void* args, uint32_t size)
+{
+    if (args && size) {
+        int ret = zjs_port_ring_buf_put(&ring_buffer, (uint16_t)id, 0, (uint32_t*)args, (uint8_t)size / 4);
+        if (ret != 0) {
+            DBG_PRINT("error putting into ring buffer, ret=%u\n", ret);
+        }
+    }
+    zjs_port_queue_put(&service_queue, cb_map[id]);
```

Comment:
This is a problem because it will overwrite the void *reserved (linked list pointer) in cb_map[id] the second time it is called for the same callback.